server/app/auth/ApiAuthenticator.java

```diff
@@ -95,14 +101,24 @@ public void validate(Credentials rawCredentials, WebContext context, SessionStor
     allowedSubnet.setInclusiveHostCount(true);
 
     if (!allowedSubnet.getInfo().isInRange(context.getRemoteAddr())) {
-      throw new BadCredentialsException(
+      unauthorized(
+          context,
           String.format(
               "IP %s not in allowed range for key ID: %s", context.getRemoteAddr(), keyId));
     }
 
     String saltedCredentialsSecret = apiKeyService.get().salt(credentials.getPassword());
     if (!saltedCredentialsSecret.equals(apiKey.getSaltedKeySecret())) {
-      throw new BadCredentialsException("Invalid secret for key ID: " + keyId);
+      unauthorized(context, "Invalid secret for key ID: " + keyId);
     }
   }
+
+  private void unauthorized(WebContext context, String cause) {
+    logger.warn(
```

Comment:
@bion  - Do we have a logger.xml to logs these? If not should we implement this  like the one you shared yesterday?

Reply:
There's already a logger configuration in the system at server/conf/logback.xml. This logger and all others in the system use the configuration there.